### DISCUSSION TITLE
Add EC variants of DSA to openssl speed -signature-algorithms benchmark command

### DISCUSSION THREAD

**Discussion by elGoonie** [openssl/openssl#23383] [Q&A] [answered]
Contrary to the CLI command `openssl speed -kem-algorithms`, no eliptic curve variants such as ECDSA or EdDSA are tested with `openssl speed -signature-algorithms`.
Especially under the new [FIPS 186-5](https://csrc.nist.gov/pubs/fips/186-5/final) standard, which considers DSA to be deprecated, EC variants should be supported in order to have better comparability of performance.

**Comment by t8m** [comment, marked as answer]
FIPS 186-5 is considering signature schemes and not KEMs. ECDSA and EdDSA is already supported by `openssl speed`

Just use `openssl speed ecdsa eddsa`

**Reply by elGoonie** [reply]
As I said, I wonder why `openssl speed -signature-algorithms` does not automatically benchmark EC variants of DSA.
Is there a reason why only RSA and DSA are benchmarked with this option?

Also, when using `openssl speed ecdsa eddsa`, only sign and verify are benchmarked. The keygen metric isn't benchmarked, unlike the `-kem-algorithms` or `-signature-algorithms` option, where we get keygen, encap/verify and decap/sign as metrics.

**Reply by t8m** [reply]
One reason for EdDSA is that that speed test benchmarks the EVP_PKEY_sign()/verify() APIs which aren't supported by EdDSA. However it would be possible to add EdDSA after PR https://github.com/openssl/openssl/pull/23240 is merged.
